### ISSUE TITLE
Users / Admins need to be able to un-delete accounts

### ISSUE THREAD

**Issue by quintessence** [mastodon/mastodon#21717]
### Pitch

When a user deletes their account on an instance they should be able to "undelete it" or the admins should be able to do so.

### Motivation

With all the Twitter exodus there are a lot of users trying different servers before they find their "home" as it were. That means it's even more common than it would've been before for a user to delete their account on a server and decide they want to reactivate their account. (We've had a few reports requesting this on Hacyderm of this nature.) Users should be able to re-claim their handle, especially if it's within a short period of time, either via the same email address they were using and/or having the instance admins be able to do it.

**Comment by coffeehabit** [comment]
Allowing accounts to be deleted saves on storage space on the server while simultaneously securing a user's privacy if they find the host undesirable to have their post history on.

**Comment by quintessence** [comment]
I agree that it is important that users (and admins) should be allowed to delete accounts. What I'm asking for here is for admins to have the ability to reverse that - to undelete an account - if the user requests it.

**Comment by danorton** [comment]
It is possible, with the tootctl `--reattach` option:

    tootctl accounts create OLDNAME --email EMAIL --reattach

**Comment by quintessence** [comment]
My understanding is even with this solution that accounts wouldn't necessarily federate correctly with other servers that would have marked the account as "deleted". Is that not the case?

**Comment by danorton** [comment]
@quintessence Can you be more specific or provide a reference?

**Comment by quintessence** [comment]
Essentially there are a couple of things:

- When a user deletes their account, that username is "locked" and it doesn't appear we can (easily) release it. So if user `example` wants to delete their account, we cannot recycle the username `example`. There are of course situations where that may not be wise to do, for identity reasons, but there are going to be many cases where users have a generic enough name that it should be released.
- When a user wishes to undelete their account, i.e. move back to a home server they previously moved away from - and they deleted, not moved, their account - they / we cannot. We were looking through past issues and found a couple that were relevant: #16626 and #16498 .

**Comment by danorton** [comment]
@quintessence  If you have access to tootctl, then you can easily release the username.

A user can return to an old username with admin assistance via tootctl.
